probatus/feature_elimination/feature_elimination.py

```diff
@@ -955,16 +935,244 @@ def __init__(
+                Train Dataset used in CV.
+
+            y_train (pd.Series):
+                Train Binary labels for X.
+
+            X_val (pd.DataFrame):
+                Validation Dataset used in CV.
+
+            y_val (pd.Series):
+                Validation Binary labels for X.
+
+            sample_weight (pd.Series, np.ndarray, list, optional):
+                array-like of shape (n_samples,) - only use if the model you're using supports
+                sample weighting (check the corresponding scikit-learn documentation).
+                Array of weights that are assigned to individual samples.
+                Note that they're only used for fitting of  the model, not during evaluation of metrics.
+                If not provided, then each sample is given unit weight.
+
+            train_index (np.array):
+                Positions of train folds samples.
+
+            val_index (np.array):
+                Positions of validation fold samples.
+
+        Raises:
+            ValueError: if the clf is not supported.
+
+        Returns:
+            dict: fit parameters
+        """
+        fit_params = {
+            "X": X_train,
+            "y": y_train,
+            "eval_set": [(X_val, y_val)],
+            "eval_metric": self.eval_metric,
+            "early_stopping_rounds": self.early_stopping_rounds,
+        }
+        if sample_weight is not None:
+            fit_params["sample_weight"] = sample_weight.iloc[train_index]
+            fit_params["eval_sample_weight"] = [sample_weight.iloc[val_index]]
+        return fit_params
+
+    def _get_fit_params_CatBoost(
+        self, X_train, y_train, X_val, y_val, sample_weight=None, train_index=None, val_index=None
+    ):
+        """Get the fit parameters for for a CatBoost Model.
+
+        Args:
+
+            X_train (pd.DataFrame):
+                Train Dataset used in CV.
+
+            y_train (pd.Series):
+                Train Binary labels for X.
+
+            X_val (pd.DataFrame):
+                Validation Dataset used in CV.
+
+            y_val (pd.Series):
+                Validation Binary labels for X.
+
+            sample_weight (pd.Series, np.ndarray, list, optional):
+                array-like of shape (n_samples,) - only use if the model you're using supports
+                sample weighting (check the corresponding scikit-learn documentation).
+                Array of weights that are assigned to individual samples.
+                Note that they're only used for fitting of  the model, not during evaluation of metrics.
+                If not provided, then each sample is given unit weight.
+
+            train_index (np.array):
+                Positions of train folds samples.
+
+            val_index (np.array):
+                Positions of validation fold samples.
+
+        Raises:
+            ValueError: if the clf is not supported.
+
+        Returns:
+            dict: fit parameters
+        """
+
```

Comment:
Can we do the import within this function instead of the top?

Reply:
I have moved the import, let me know if we can resolve this comment